src/dolphin/iniFile.ts

```diff
@@ -0,0 +1,278 @@
+      if (line[0] === "[") {
+        //console.log(line, line[0]);
+        const endpos = line.indexOf("]");
+        if (endpos !== -1) {
+          //we have a new section
+          const sub = line.substr(1, endpos - 1);
+          //console.log(sub);
+          current_section = this.getOrCreateSection(sub);
+          //console.log(current_section);
+        }
+      } else {
+        if (current_section !== undefined) {
+          const [key, value] = this.parseLine(line);
+
+          // Lines starting with '$', '*' or '+' are kept verbatim.
+          // Kind of a hack, but the support for raw lines inside an
+          // INI is a hack anyway.
+          if (
+            (key === null && value === null) ||
+            (line.length !== 0 && ["$", "+", "*"].some((val) => line[0] === val))
+          ) {
+            current_section.m_lines.push(line);
+          } else if (key !== null && value !== null) {
+            current_section.set(key, value);
+          }
+        }
+      }
+    }
+
+    return true;
+  }
+
+  public save(filePath: string): boolean {
+    const out = fs.createWriteStream(filePath);
+
+    out.on("error", (e) => {
+      console.log("failed to write file with error", e);
+      alert(e);
+    });
+
+    this.sections.forEach((section) => {
+      // originally section.name was only written if the section was non-empty,
+      // but that goes against us wanting to always show the Gecko section
+      out.write(`[${section.name}]\n`);
+
+      if (section.keys_order.length === 0) {
+        section.m_lines.forEach((line) => {
+          out.write(`${line}\n`);
+        });
+        out.write("\n");
+      } else {
+        section.keys_order.forEach((kvit) => {
+          const value = section.values.get(kvit);
+          out.write(`${kvit}=${value}\n`);
+        });
+        out.write("\n");
+      }
+    });
+
+    out.end();
+    out.close();
+
+    return true;
+  }
+}
+
+/**
+ * The Section class
+ */
+export class Section {
+  public name: string;
+  public keys_order: string[];
+  public m_lines: string[];
+  public values: Map<string, string>;
+
+  public constructor(_name: string) {
+    this.name = _name;
+    this.keys_order = [];
+    this.m_lines = [];
+    this.values = new Map();
+  }
+
+  /**Differs from IniFile.cpp by:
+   * passes key by value rather than address
+   */
+  public set(key: string, new_value: string): void {
+    const newKey = !this.values.has(key);
+    if (newKey) {
+      this.keys_order.push(key);
+    }
+    this.values.set(key, new_value);
+  }
+
+  //TODO work around pass by reference
+  // no idea what default value is for
+  public get(key: string, default_value: string): string {
+    const value = this.values.get(key);
+
+    if (value !== undefined) {
+      return value;
+    }
+
+    return default_value;
+  }
+
+  public exists(key: string): boolean {
+    return this.values.get(key) !== undefined;
+  }
+
+  public delete = (key: string): boolean => {
```

Comment:
```suggestion
  public delete(key: string): boolean {
```
We should probably be consistent with the method declarations.